src/main/java/org/takes/tk/TkProxy.java

```diff
@@ -0,0 +1,173 @@
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.Take;
+import org.takes.rq.RqHeaders;
+import org.takes.rq.RqHref;
+import org.takes.rq.RqLengthAware;
+import org.takes.rq.RqMethod;
+import org.takes.rq.RqPrint;
+import org.takes.rs.RsWithBody;
+import org.takes.rs.RsWithHeaders;
+import org.takes.rs.RsWithStatus;
+
+/**
+ * Take that proxies requests to another destination.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Dragan Bozanovic (bozanovicdr@gmail.com)
+ * @version $Id$
+ * @since 0.25
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class TkProxy implements Take {
+
+    /**
+     * Target host to which requests are forwarded.
+     */
+    private final transient String host;
+
+    /**
+     * Ctor.
+     *
+     * <p>The {@code target} parameter takes the destination URL
+     *  to which requests are proxied. Some valid examples:
+     *  <li>example.com
+     *  <li>www.example.com
+     *  <li>example.com:8080
+     *  <li>example.com/x/y
+     * @param target Target to which requests are forwarded
+     */
+    public TkProxy(final String target) {
+        this.host = target;
+    }
+
+    @Override
+    public Response act(final Request req) throws IOException {
+        final RqHref.Base base = new RqHref.Base(req);
+        final String home = new RqHref.Smart(base).home().bare();
+        final String dest = String.format(
+            "http://%s/%s",
+            this.host,
+            base.href().toString().substring(home.length())
+        );
+        return this.response(home, dest, this.request(req, dest).fetch());
+    }
+
+    /**
+     * Creates the request to be forwarded to the target host.
+     *
+     * @param req Original request
+     * @param dest Destination URL
+     * @return Request to be forwarded
+     * @throws IOException If some problem inside
+     */
+    private com.jcabi.http.Request request(final Request req,
+        final String dest) throws IOException {
+        final String method = new RqMethod.Base(req).method();
+        com.jcabi.http.Request proxied = new JdkRequest(dest)
+            .method(method);
+        final RqHeaders.Base headers = new RqHeaders.Base(req);
+        for (final String name : headers.names()) {
+            if ("content-length".equals(name.toLowerCase(Locale.ENGLISH))) {
+                continue;
+            }
+            if (isHost(name)) {
+                proxied = proxied.header(name, this.host);
+                continue;
+            }
+            for (final String value : headers.header(name)) {
+                proxied = proxied.header(name, value);
+            }
+        }
+        if (Arrays.asList("POST", "PUT").contains(method)) {
```

Comment:
@yegor256 Good point. Actually, I tried to implemented it that way initially, but I encountered the issue that makes all attempts to read from the body input stream from `GET` requests to block forever.

I intended to write a separate issue for that later (actually I did it now, please see #438), and I added support for proxying only POST and PUT request bodies (which should cover most of the cases in practice). However, I agree that I should have added at least a puzzle directly in this PR.

I changed my implementation to check for the presence of `Content-Length` or `Transfer-Encoding` headers before attempting to read the request body, as described [here](http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.3):
>The presence of a message-body in a request is signaled by the inclusion of a Content-Length or Transfer-Encoding header field in the request's message-headers.

What do you think?